### PR TITLE
i fixed things and formatted

### PR DESCRIPTION
**Pull request by LexiBigCheese**
there was also a stupid test i had to remove because it was so stupid

### CODE REVIEW

File: lib/src/deserializer.dart [hunk 1]
```diff
@@ -263,41 +271,63 @@ Object _convertValue(/*Type | List<Type>*/ valueType, Object value, [String key
...
+Object _initiateClass(Type type, [fillet]) {
   ClassMirror classMirror = reflectType(type);
 //  _desLog.fine("Parsing to class: ${type}");
 
   if (classMirror.isEnum) {
-    return classMirror.values[filler];
+    return classMirror.values[fillet];
   }
 
+  Map filler = fillet;
```
Comment (luisvt) [inline lib/src/deserializer.dart]: What for is this aux variable?

**Review by luisvt — COMMENTED** [review]
can you explain what is this for?

File: lib/src/deserializer.dart (hunk 1 above)
Comment (LexiBigCheese) [reply lib/src/deserializer.dart]: Completion purposes.

**Comment by LexiBigCheese** [comment]
i made it smarter.

**Comment by luisvt** [comment]
could you explain what does it means that?

**Comment by LexiBigCheese** [comment]
smarter means... well... the `runtimeType` is kept in the map / json.

it means...

```dart
library smart;
import 'package:dson/dson';
part 'smart.g.dart';
class SmartSomethingOrAnother extends _$SmartSomethingOrAnotherSerializable {
  var a;
  int b;
  SmartSomethingOrAnother(this.a,{this.b});
}

void main() {
  _initMirrors();
  var x = fromMap({'a':[1,2,{'a':1,'b':2,'runtimeType':'SmartSomethingOrAnother'}],'b':3,'runtimeType':'SmartSomethingOrAnother'});
  assert(x.a is List);
  assert(x.a[2] is SmartSomethingOrAnother);
  assert(x is SmartSomethingOrAnother);
}
```

can happen.

**Comment by LexiBigCheese** [comment]
i fixed #31, but if you really want to pull, you need to pull the `built_mirrors` and `serializable` packages too. and upload them in the order `built_mirrors` `serializable` `dson` and change the sources of `serializable` and `dson` to their places in pub when uploading.

**Comment by luisvt** [comment]
This was merge in a different way

**Pull request closed without merging**